src/OpenTracing/Span.php

```diff
@@ -0,0 +1,130 @@
+     * @return $this
+     */
+    abstract public function setOperationName($operationName);
+
+    /**
+     * Indicates that the work represented by this span has been completed
+     * or terminated, and is ready to be sent to the Reporter.
+     *
+     * If any tags / logs need to be added to the span, it should be done
+     * before calling finish(), otherwise they may be ignored.
+     *
+     * @param float $finishTime
+     */
+    abstract public function finish($finishTime = null);
+
+    /**
+     * Attaches a key/value pair to the span.
+     *
+     * The set of supported value types is implementation specific. It is the
+     * responsibility of the actual tracing system to know how to serialize
+     * and record the values.
+     *
+     * If the user calls set_tag multiple times for the same key,
+     * the behavior of the tracer is undefined, i.e. it is implementation
+     * specific whether the tracer will retain the first value, or the last
+     * value, or pick one randomly, or even keep all of them.
+     *
+     * @param string $key
+     * @param mixed $value
+     * @return $this
+     */
+    abstract public function setTag($key, $value);
+
+    /**
+     * Logs an event against the span, with the current timestamp.
+     *
+     * @param string $event
+     * @param array $payload
+     * @return $this
+     */
+    abstract public function logEvent($event, $payload = null);
+
+    /**
+     * Records a generic Log event at an arbitrary timestamp.
+     *
+     * @param float $timestamp
+     * @param string $event
+     * @param array $payload
+     * @return $this
+     */
+    abstract public function log($timestamp, $event, $payload = null);
+
+    /**
+     * Stores a Trace Attribute in the span as a key/value pair.
+     *
+     * Enables powerful distributed context propagation functionality where
+     * arbitrary application data can be carried along the full path of
+     * request execution throughout the system.
+     *
+     * Note 1: attributes are only propagated to the future (recursive)
+     * children of this Span.
+     *
+     * Note 2: attributes are sent in-band with every subsequent local and
+     * remote calls, so this feature must be used with care.
+     *
+     * Note 3: keys are case-insensitive, to allow propagation via HTTP
+     * headers. Keys must match regexp `(?i:[a-z0-9][-a-z0-9]*)`
+     *
+     * @param string $key
+     * @param mixed $value
+     * @return $this
+     */
+    abstract public function setTraceAttribute($key, $value);
+
+    /**
+     * Retrieves value of the Trace Attribute with the given key.
+     *
+     * Returns null if key doesn't exist.
+     *
+     * Key is case-insensitive.
+     *
+     * @param string $key
+     * @return mixed
+     */
+    abstract public function getTraceAttribute($key);
+
+    /**
+     * Provides access to the Tracer that created this Span.
+     *
+     * @return Tracer
+     */
+    public function getTracer() {
```

Comment:
I think this should be abstract as well, otherwise its forcing `$this->tracer` to exist, maybe implementations don't need that.